test_project/views/frontend/widget_data.py

```diff
@@ -60,54 +62,54 @@ def handle_request(self, request):
 
         while True:
             # list
-            widget.data = {'list': []}
+            widget.widget_data = {'list': []}
```

Comment:
Just nitpicking:
It would probably be better style to already set this value in ``TestWidget.__init__()``. This way it will always be available when calling ``TestWidget.update_state()``.

(But this is not relevant to the test case here.)

Reply:
Ack! I added this view to my refactoring list, but i don't want to change it to much in this pr.